avocado/core/job.py

```diff
@@ -462,6 +465,7 @@ def _run(self):
         """
         self._setup_job_results()
         self.__start_job_logging()
+        self.job_pre_post_dispatcher.map_methods('pre', self)
```

Comment:
One more think to be clarified, do we want to run pre-job plugins on `--dry-run`?